### ISSUE TITLE
[Phase 8] Lennard-Jones Extension

### ISSUE THREAD

**Issue by Sakeeb91** [Sakeeb91/diffusion-boltzmann-sampler#10]
## Objective

Extend system to continuous Lennard-Jones fluid particles.

## Deliverables

- [ ] `LennardJonesSystem` with energy and force computation
- [ ] MLP score network with periodic embeddings
- [ ] `LJVisualizer` component for particle positions
- [ ] Radial distribution function g(r) analysis
- [ ] Training pipeline for L-J system

## Files to Create

| File | Description |
|------|-------------|
| `backend/ml/systems/lennard_jones.py` | L-J potential and forces |
| `backend/ml/models/score_network_continuous.py` | MLP for continuous systems |
| `frontend/src/components/LJVisualizer.tsx` | Particle visualization |
| `backend/ml/analysis/rdf.py` | Radial distribution function |

## Core Implementation

```python
# backend/ml/systems/lennard_jones.py
import torch

class LennardJonesSystem:
    """2D Lennard-Jones fluid with periodic boundaries."""

    def __init__(self, n_particles: int, box_size: float,
                 epsilon: float = 1.0, sigma: float = 1.0, cutoff: float = 2.5):
        self.n_particles = n_particles
        self.box_size = box_size
        self.epsilon = epsilon
        self.sigma = sigma
        self.cutoff = cutoff * sigma

    def minimum_image(self, dr: torch.Tensor) -> torch.Tensor:
        """Apply minimum image convention."""
        return dr - self.box_size * torch.round(dr / self.box_size)

    def pairwise_distances(self, positions: torch.Tensor):
        """Compute all pairwise distances."""
        r_i = positions[..., :, None, :]
        r_j = positions[..., None, :, :]
        dr = self.minimum_image(r_i - r_j)
        dist = torch.sqrt((dr ** 2).sum(dim=-1) + 1e-10)
        return dist, dr

    def energy(self, positions: torch.Tensor) -> torch.Tensor:
        """Compute total L-J energy."""
        dist, _ = self.pairwise_distances(positions)
        mask = (dist > 0) & (dist < self.cutoff)
        dist_m = torch.where(mask, dist, torch.ones_like(dist))

        r6 = (self.sigma / dist_m) ** 6
        r12 = r6 ** 2
        potential = 4 * self.epsilon * (r12 - r6)
        potential = torch.where(mask, potential, torch.zeros_like(potential))

        return 0.5 * potential.sum(dim=(-1, -2))

    def forces(self, positions: torch.Tensor) -> torch.Tensor:
        """Compute forces: F = -∇E."""
        dist, dr = self.pairwise_distances(positions)
        mask = (dist > 0) & (dist < self.cutoff)
        dist_m = torch.where(mask, dist, torch.ones_like(dist))

        r6 = (self.sigma / dist_m) ** 6
        r12 = r6 ** 2
        f_mag = 24 * self.epsilon / dist_m * (2 * r12 - r6)
        f_mag = torch.where(mask, f_mag, torch.zeros_like(f_mag))

        unit = dr / (dist_m[..., None] + 1e-10)
        return (f_mag[..., None] * unit).sum(dim=-2)

    def score(self, positions: torch.Tensor, temperature: float):
        """Score = F / kT."""
        return self.forces(positions) / temperature
```

```typescript
// frontend/src/components/LJVisualizer.tsx
export const LJVisualizer: React.FC<{
  positions: [number, number][];
  boxSize: number;
}> = ({ positions, boxSize }) => (
  <Plot
    data={[{
      x: positions.map(p => p[0]),
      y: positions.map(p => p[1]),
      mode: 'markers',
      marker: { size: 10, color: '#3b82f6' },
    }]}
    layout={{
      width: 400, height: 400,
      xaxis: { range: [0, boxSize], title: 'x' },
      yaxis: { range: [0, boxSize], title: 'y', scaleanchor: 'x' },
    }}
  />
);
```

## Technical Challenges

1. **Periodic boundaries** - Minimum image convention for distances
2. **Cutoff radius** - Truncate potential at r_c to avoid O(N²) per step
3. **Softening** - Avoid 1/r singularity at small r
4. **Continuous score** - MLP architecture different from U-Net

## Verification

- Energy conserved in NVE simulation
- g(r) shows correct liquid structure
- Score matches numerical gradient of energy
- Neural sampler reproduces g(r) from MCMC

## Definition of Done

- L-J system runs without numerical issues
- Radial distribution function computed correctly
- Frontend displays particles with periodic wrap
- Diffusion sampler works for continuous coordinates

## Dependencies

- Phase 4: Diffusion Sampling (adapt for continuous)
- Phase 5: Frontend Core

## Related

Meta issue: #1

**Comment by Sakeeb91** [comment]
Implemented Lennard-Jones extension with 15 atomic commits.

## Changes
- `LennardJonesSystem` with energy, forces, and score computation
- `ContinuousScoreNetwork` (MLP) with periodic position embeddings
- `LangevinDynamics` and `UnderdampedLangevin` samplers
- Radial distribution function g(r) analysis
- `LJVisualizer` frontend component
- REST API endpoints at `/sample/lj/*`

## Files Created
- `backend/ml/systems/lennard_jones.py`
- `backend/ml/models/score_network_continuous.py`
- `backend/ml/samplers/langevin.py`
- `backend/ml/analysis/rdf.py`
- `backend/api/routes/lj_sampling.py`
- `frontend/src/components/LJVisualizer.tsx`
- `backend/tests/test_lennard_jones.py`
- `backend/tests/test_rdf.py`

## Tests
All 353 backend tests pass, including 18 new L-J tests and 11 RDF tests.
